Add initial files.

davidlehn opened the pull request.

davidlehn committed:
Add initial files.

kezike committed:
Fix minor misspelling.

kezike commented on js/index.js:
```diff
@@ -0,0 +1,33 @@
+module.exports = {
+  CONTEXT: context,
+  CONTEXT_URL: constants.CONTEXT_URL,
+  appContextMap,
+  constants,
+  contexts
```
@davidlehn For some reason, these three variables are not able to be imported from this library. Here is the kind of error that we get:
```
SyntaxError: Named export 'contexts' not found. The requested module '@digitalbazaar/vdl-aamva-context' is a CommonJS module, which may not support all module.exports as named exports.
```
The only thing that confuses me here is that this error is _only_ visible with these variables but not the other two. We really only need the other two for our (and most) use cases anyway, but I wonder if you can see why there was an issue with these variables and if it's worthwhile to fix it.

davidlehn replied:
The CJS/ESM story here is a bit convoluted due to history of these context modules.  I may have fixed it with some more explicit ESM bits.  But also see: https://github.com/digitalbazaar/vdl-aamva-context/pull/2 that flips this to a module that generates CommonJS.

kezike committed:
Clean up readme.

dlongley commented on js/index.js:
```diff
@@ -0,0 +1,33 @@
+  // default context CBOR-LD codec id
+  // value between 0x0 and 0x7FFF for globally registered term codec values
+  // or >= 0x8000 for app-specific local terms
+  CBORLD_VALUE: 0x0
```
We should probably pick a proper value here or not expose this -- otherwise someone will use it. (I'm presuming `0x0` isn't a proper value here).

davidlehn replied:
Zero is reserved.  We should probably make zero in particular mean to not use those magic values.

kezike replied:
@davidlehn I noticed that this discussion was revisited in ***. Do we have an idea about what this value will be for VDL AAMVA?

dlongley reviewed (approved):
Approving -- with the assumption we'll not expose the `CBORLD_VALUE` or pick a value for it. It looks like it's set on the default of `0x0`, which I presume is already taken / not a good choice.

davidlehn commented on README.md:
```diff
@@ -1 +1,114 @@
+
+See also (related specs):
+
+* https://w3c-ccg.github.io/vdl-vocab
```
This was changed to remove trailing `/` but that server redirects to URL with the `/`.

kezike replied:
Fixed ✅

davidlehn commented on README.md:
```diff
@@ -1 +1,114 @@
+- BSD 3-Clause © Digital Bazaar
+- See the [LICENSE](./LICENSE) file for details.
+
+[CBOR-LD]: https://digitalbazaar.github.io/cbor-ld-spec
```
This was changed to remove trailing `/` but that server redirects to URL with the `/`.

kezike replied:
Fixed ✅

davidlehn committed:
Improve ESM compatibility.

- No need to distribute CommonJS main.js.
- Add `import` test.
- Add `exports` to package.json.

davidlehn committed:
Resolve merge conflict with initial branch.

kezike committed:
Update to latest VDL AAMVA context from 11/3.

kezike committed:
Move constants to their own file.

davidlehn commented on package-lock.json:
Please remove from repo and history.

davidlehn commented on js/context.js:
```diff
@@ -5,12 +5,11 @@
 
-module.exports =
-{
+module.exports = {
```
The odd style here was intentional in order to make the entire JSON block easy to copy.  Is there a reason to change it?

kezike replied:
Oh, I did it because that is the style I am use to, but I can change it back.

davidlehn commented:
@kezike Why were constants moved into their own file?
I ask because merging them into one file was an attempt to simplify an older context package style.

kezike committed:
Fix linting.

kezike commented:
> @kezike Why were constants moved into their own file? I ask because merging them into one file was an attempt to simplify an older context package style.

@davidlehn Oh, I did it because I noticed it from other context libraries, but I can change it back.

kezike committed:
Add back slashes for spec URLs.

kezike committed:
Remove CommonJS artifacts.

kezike committed:
Remove duplicate key in .eslintrc.cjs.

kezike committed:
Remove package-lock.json from git history.

kezike committed:
Import constants from separate file.

kezike committed:
Define constants in index file.

kezike committed:
Rearrange lines slightly in README.md.

kezike committed:
Fix copyright year.

kezike committed:
Add minor cosmetics changes.

kezike commented:
@dlongley @davidlehn Bumping for final review ☝️

davidlehn committed:
Fix module fields.

- Use full relative paths.
- Use ESM code for "module".

davidlehn committed:
Update dependencies.

davidlehn committed:
Update action versions.

davidlehn committed:
Update copyright years.
